### PR TITLE
Add podcasts guid, set by uuidv5, add to rss

### LINKED ISSUE #1110: Introduce a guid to use for identifying a podcast
support the podcasting 2.0 `podcasts:guid` in the rss feed

https://github.com/Podcastindex-org/podcast-namespace/blob/main/docs/1.0.md#guid

and this would be a less predicatable/seqeuntual id to use in public - perhaps for use in integrations with other systems

### PR DESCRIPTION
**Pull request by kookster**
fixes #1110

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -443,8 +444,8 @@
   end
 
   create_table "tasks", id: :serial, force: :cascade do |t|
-    t.integer "owner_id"
     t.string "owner_type"
+    t.integer "owner_id"
```
Comment (cavis) [inline db/schema.rb:448]: Hrm, couple of unrelated changes in here.  Are you working off a prod backup?

File: db/migrate/20240928150204_add_guid_to_podcasts.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddGuidToPodcasts < ActiveRecord::Migration[7.2]
+  def change
+    add_column :podcasts, :guid, :string
```
Comment (cavis) [inline db/migrate/20240928150204_add_guid_to_podcasts.rb:3]: Any reason not to just generate these for all existing Podcasts?

**Review by cavis — APPROVED** [review]
👍 looks good to me.  I do wonder if we shouldn't just add these to all existing podcasts.

File: db/schema.rb (hunk 1 above)
Comment (kookster) [reply db/schema.rb:448]: it just changed the order of the columns, not the type or anything - is that a problem?

Comment (kookster) [reply db/schema.rb:448]: (I was not working off a copy of prod when I ran these migrations.)

Comment (cavis) [reply db/schema.rb:448]: Not a problem - it'll just flicker back the next time someone runs a migration locally.

File: db/migrate/20240928150204_add_guid_to_podcasts.rb (hunk 2 above)
Comment (kookster) [reply db/migrate/20240928150204_add_guid_to_podcasts.rb:3]: Yes, and we could do that at the console at any time, I just didn't force it in this PR

**Review by cavis — APPROVED** [review]
👍
